web/src/main/java/org/cbioportal/web/parameter/GeneFilter.java

```diff
@@ -0,0 +1,125 @@
+package org.cbioportal.web.parameter;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import javax.validation.constraints.AssertTrue;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+public class GeneFilter implements Serializable {
+
+    private Set<String> molecularProfileIds;
+    private List<List<String>> geneQueries;
+
+    private final String GENE_QUERY_PATTERN = "^(\\w+)[\\s]*?(?:\\:(?:[\\s]*(?:(AMP)|(HOMDEL))\\b)+)?$";
```

Comment:
Is this patter coming from the frontend?  Do you think it makes sense to have all OQL related regex to be placed in a same file.

Reply:
no. this is specific for study filter. we just allow `AMP` and `HOMDEL`

Reply:
Does this pattern have to get updated with introduction of CNA enum?

Reply:
probably not. As only `AMP` and `HOMDEL` are allowed here